<!DOCTYPE html>
<html lang="en"><head>
<script src="R_dada2_tutorial_files/libs/clipboard/clipboard.min.js"></script>
<script src="R_dada2_tutorial_files/libs/quarto-html/tabby.min.js"></script>
<script src="R_dada2_tutorial_files/libs/quarto-html/popper.min.js"></script>
<script src="R_dada2_tutorial_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="R_dada2_tutorial_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="R_dada2_tutorial_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="R_dada2_tutorial_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="R_dada2_tutorial_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.97">

  <title>r_dada2_tutorial</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="R_dada2_tutorial_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="R_dada2_tutorial_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
      }
    pre.numberSource { margin-left: 3em;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2;  }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #f07178; background-color: #2a0f15; font-weight: bold; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #00e0e0; } /* Attribute */
    code span.bn { color: #d4d0ab; } /* BaseN */
    code span.bu { color: #abe338; } /* BuiltIn */
    code span.cf { color: #ffa07a; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffd700; } /* Constant */
    code span.co { color: #f8f8f2; font-style: italic; } /* Comment */
    code span.cv { color: #ffd700; } /* CommentVar */
    code span.do { color: #f8f8f2; } /* Documentation */
    code span.dt { color: #ffa07a; } /* DataType */
    code span.dv { color: #d4d0ab; } /* DecVal */
    code span.er { color: #f07178; text-decoration: underline; } /* Error */
    code span.ex { color: #00e0e0; font-weight: bold; } /* Extension */
    code span.fl { color: #d4d0ab; } /* Float */
    code span.fu { color: #ffa07a; } /* Function */
    code span.im { color: #abe338; } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; font-weight: bold; } /* Keyword */
    code span.op { color: #ffa07a; } /* Operator */
    code span.ot { color: #00e0e0; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.re { color: #00e0e0; background-color: #f8f8f2; } /* RegionMarker */
    code span.sc { color: #abe338; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #00e0e0; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #dcc6e0; } /* Warning */
  </style>
  <link rel="stylesheet" href="R_dada2_tutorial_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="R_dada2_tutorial_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="R_dada2_tutorial_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="R_dada2_tutorial_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="R_dada2_tutorial_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  
  <script src="R_dada2_tutorial_files/libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
  <link href="R_dada2_tutorial_files/libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
  <script src="R_dada2_tutorial_files/libs/datatables-binding-0.26/datatables.js"></script>
  <script src="R_dada2_tutorial_files/libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="R_dada2_tutorial_files/libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
  <link href="R_dada2_tutorial_files/libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
  <script src="R_dada2_tutorial_files/libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
  <link href="R_dada2_tutorial_files/libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
  <script src="R_dada2_tutorial_files/libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">


<section class="slide level2">

<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-1_37cbe3bcfa7ac9af07a2375facd6f982">

</div>
<p><span class="title absolute" style="bottom: 550px; ">R course</span></p>
<p><span class="presenter absolute" style="bottom: 300px; "><strong>Daniel Vaulot</strong></span></p>
<p><span class="absolute smaller" style="bottom: 250px; ">2023-01-18</span></p>
<p><img data-src="img/institutions/logo_ecodim.png" class="absolute" style="left: 600px; bottom: 20px; width: 200px; "></p>
<p><img data-src="img/institutions/logo_SBR.png" class="absolute" style="left: 0px; bottom: 20px; height: 100px; "></p>
<p><img data-src="img/institutions/NTU-Logo-full-colour.png" class="absolute" style="bottom: 20px; right: 0px; height: 100px; "></p>
<p><img data-src="img/R/R-logo.png" class="absolute" style="top: 50px; right: 50px; height: 300px; "></p>
<p><span class="subtitle absolute" style="bottom: 450px; ">05 - Metabarcode processing with dada2</span></p>
</section>
<section id="introduction" class="slide level2">
<h2>Introduction</h2>
<p>This tutorial explain how to process Illumina data with the Dada2 suite as implemented in R (dada2 is also implemented in Qiime).</p>
<p>It is adapted from the <a href="https://benjjneb.github.io/dada2/tutorial.html">Dada2 tutorial</a></p>
<h3 id="directory-structure">Directory structure</h3>
<p>Relative to the main directory from GitHub</p>
<ul>
<li><strong>../fastq</strong> : fastq files</li>
<li><strong>../fastq_filtered</strong> : fastq files after filtration</li>
<li><strong>../qual_pdf</strong> : qual pdf files</li>
<li><strong>../dada2</strong> : dada2 processed files</li>
<li><strong>../databases</strong> : <a href="https://github.com/pr2database/pr2database/releases/">PR2 database file</a></li>
<li><strong>../blast</strong> : BLAST files output</li>
<li><strong>../R</strong> : This tutorial for Illumina files</li>
</ul>
</section>
<section id="downloads" class="slide level2">
<h2>Downloads</h2>
<p>Install the following software :</p>
<ul>
<li><p><a href="https://cran.r-project.org/index.html">R</a></p></li>
<li><p><a href="https://www.rstudio.com/products/rstudio/download/#download">R studio</a></p></li>
</ul>
<p>Download and install the following libraries by running under R studio the following lines</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-2_35a3955504ae57f4cebd0905713bbf83">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"readr"</span>)     <span class="co"># To read and write files</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"readxl"</span>)    <span class="co"># To read excel files</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"dplyr"</span>)     <span class="co"># To manipulate dataframes</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tibble"</span>)    <span class="co"># To work with data frames</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyr"</span>)     <span class="co"># To work with data frames</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"stringr"</span>)   <span class="co"># To manipulate strings</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)   <span class="co"># To do plots</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"Biobase"</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"phyloseq"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-used" class="slide level2">
<h2>Data used</h2>
<p><img data-src="img/cruise/carbom_cruise.png" class="absolute" style="top: 100px; right: 50px; width: 500px; "></p>
<div class="columns">
<div class="column" style="width:60%;">
<p>The samples originate from the CARBOM cruise (2013) off Brazil.</p>
<p>Samples have been sorted by flow cytometry and 3 genes have been PCR amplified :</p>
<ul>
<li>18S rRNA - V4 region</li>
<li>16S rNA with plastid</li>
<li><em>nifH</em></li>
</ul>
<p>The PCR products have been sequenced by 1 run of Illumina 2*250 bp. The data consist of the picoplankton samples from one transect and fastq files have been subsampled with 1000 sequences per sample.</p>
</div>
</div>
<h3 id="references">References</h3>
<ul>
<li>Gerikas Ribeiro C, Marie D, Lopes dos Santos A, Pereira Brandini F, Vaulot D. (2016). Estimating microbial populations by flow cytometry: Comparison between instruments. Limnol Oceanogr Methods 14:750–758.</li>
<li>Gerikas Ribeiro C, Lopes dos Santos A, Marie D, Brandini P, Vaulot D. (2018). Relationships between photosynthetic eukaryotes and nitrogen-fixing cyanobacteria off Brazil. ISME J in press.</li>
<li>Gerikas Ribeiro C, Lopes dos Santos A, Marie D, Helena Pellizari V, Pereira Brandini F, Vaulot D. (2016). Pico and nanoplankton abundance and carbon stocks along the Brazilian Bight. PeerJ 4:e2587.</li>
</ul>
</section>
<section>
<section id="set-up" class="title-slide slide level1 center">
<h1>Set-up</h1>

</section>
<section id="load-the-necessary-libraries" class="slide level2">
<h2>Load the necessary libraries</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-3_a4f87d6a16b6889d04a75fdfcf8422de">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"dada2"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"phyloseq"</span>) </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"tibble"</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"readxl"</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"readr"</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">"kableExtra"</span>) <span class="co"># necessary for nice table formatting with knitr</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-up-directories" class="slide level2">
<h2>Set up directories</h2>
<p>Create directories that will be used to stored the files at the different stage of the processing</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-4_63b3e4fc001c008b997f8e25edc44800">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change the following line to the path where you unzipped the tutorials</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  fastq_dir <span class="ot">&lt;-</span>    <span class="st">"../fastq/"</span>           <span class="co"># fastq directory</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  filtered_dir <span class="ot">&lt;-</span> <span class="st">"../fastq_filtered/"</span>  <span class="co"># fastq filtered</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  qual_dir <span class="ot">&lt;-</span>     <span class="st">"../qual_pdf/"</span>        <span class="co"># qual pdf</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  dada2_dir <span class="ot">&lt;-</span>    <span class="st">"../dada2/"</span>           <span class="co"># dada2 results</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  blast_dir <span class="ot">&lt;-</span>    <span class="st">"../blast/"</span>           <span class="co"># blast2 results</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  database_dir <span class="ot">&lt;-</span> <span class="st">"../databases/"</span>       <span class="co"># databases</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(filtered_dir)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(qual_dir)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(dada2_dir)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(blast_dir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-variables" class="slide level2">
<h2>Setup variables</h2>
<h4 id="primers">Primers</h4>
<p>Note that the primers are degenerated.</p>
<p>Dada2 has an option to remove primers (<code>FilterandTrim</code>) but this function will not accept degeneracy.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-5_03b05e50eb44d0616a78cedc0915df42">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>  primer_set_fwd <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"CCAGCAGCCGCGGTAATTCC"</span>, <span class="st">"CCAGCACCCGCGGTAATTCC"</span>, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"CCAGCAGCTGCGGTAATTCC"</span>, <span class="st">"CCAGCACCTGCGGTAATTCC"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  primer_set_rev <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ACTTTCGTTCTTGATYRATGA"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  primer_length_fwd <span class="ot">&lt;-</span> <span class="fu">str_length</span>(primer_set_fwd[<span class="dv">1</span>]) </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  primer_length_rev <span class="ot">&lt;-</span> <span class="fu">str_length</span>(primer_set_rev[<span class="dv">1</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<h4 id="pr2-tax-levels">PR2 tax levels</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-6_614677440b89fb297c10df1662b50ce4">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>  PR2_tax_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Supergroup"</span>,<span class="st">"Division"</span>, <span class="st">"Class"</span>, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section>
<section>
<section id="examine-the-fastq-files" class="title-slide slide level1 center">
<h1>Examine the fastQ files</h1>

</section>
<section id="construct-a-list-of-the-fastq-files" class="slide level2">
<h2>Construct a list of the fastq files</h2>
<p>It is assumed that the sample names are at the start of file name and separated by _.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-7_3e52e05010294046ec674f5c42e06a30">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get a list of all fastq files in the ngs directory and separate R1 and R2</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  fns <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(fastq_dir, <span class="at">full.names =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  fns <span class="ot">&lt;-</span> fns[<span class="fu">str_detect</span>( <span class="fu">basename</span>(fns),<span class="st">".fastq"</span>)]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  fns_R1 <span class="ot">&lt;-</span> fns[<span class="fu">str_detect</span>( <span class="fu">basename</span>(fns),<span class="st">"R1"</span>)]</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  fns_R2 <span class="ot">&lt;-</span> fns[<span class="fu">str_detect</span>( <span class="fu">basename</span>(fns),<span class="st">"R2"</span>)]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  sample.names <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="fu">basename</span>(fns_R1), <span class="at">pattern =</span> <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>) </span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  sample.names <span class="ot">&lt;-</span> sample.names[,<span class="dv">1</span>]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "120p" "121p" "122p" "125p" "126p" "140p" "141p" "142p" "155p" "156p"
[11] "157p" "165p" "166p" "167p"</code></pre>
</div>
</div>
</section>
<section id="compute-number-of-paired-reads" class="slide level2">
<h2>Compute number of paired reads</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/fastq_reads_ee614115c622328993e3275e9dee69e8">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create an empty data frame  </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()  </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># loop through all the R1 files (no need to go through R2 which should be the same) </span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fns_R1)) { </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># use the dada2 function fastq.geometry</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>      geom <span class="ot">&lt;-</span> <span class="fu">fastq.geometry</span>(fns_R1[i])</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># extract the information on number of sequences and file name </span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>      df_one_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span> (<span class="at">n_seq=</span>geom[<span class="dv">1</span>], <span class="at">file_name=</span><span class="fu">basename</span>(fns_R1[i]) )</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add one line to data frame</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(df, df_one_row)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  } </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<h3 id="display-results">Display results</h3>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/fastq_reads_results_75be5a7e5af51f1f0035b2adba980114">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># display number of sequences</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>(df)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the histogram with number of sequences</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>n_seq)) <span class="sc">+</span> </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_histogram</span>( <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">binwidth =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">2000</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f71e6fe4a8262706a134" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f71e6fe4a8262706a134">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],[1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000],["120p_S39_R1.subsample.fastq","121p_S57_R1.subsample.fastq","122p_S4_R1.subsample.fastq","125p_S22_R1.subsample.fastq","126p_S40_R1.subsample.fastq","140p_S5_R1.subsample.fastq","141p_S23_R1.subsample.fastq","142p_S41_R1.subsample.fastq","155p_S59_R1.subsample.fastq","156p_S6_R1.subsample.fastq","157p_S24_R1.subsample.fastq","165p_S42_R1.subsample.fastq","166p_S60_R1.subsample.fastq","167p_S7_R1.subsample.fastq"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_seq<\/th>\n      <th>file_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<img data-src="R_dada2_tutorial_files/figure-revealjs/fastq_reads_results-1.png" width="768" class="r-stretch"></section>
<section id="plot-quality-for-reads" class="slide level2">
<h2>Plot quality for reads</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/fastq_quality_df3ed2b85af3bb846f9b0a48ea5a9a4a">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a> <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fns)) { </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use dada2 function to plot quality</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> <span class="fu">plotQualityProfile</span>(fns[i])</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only plot on screen for first 2 files  </span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (i <span class="sc">&lt;=</span> <span class="dv">2</span>) {<span class="fu">print</span>(p1)}</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save the file as a pdf file (uncomment to execute)</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    p1_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(qual_dir, <span class="fu">basename</span>(fns[i]),<span class="st">".qual.pdf"</span>)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>( <span class="at">plot=</span>p1, <span class="at">filename=</span> p1_file,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">device =</span> <span class="st">"pdf"</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">15</span>, <span class="at">scale=</span><span class="dv">1</span>, <span class="at">units=</span><span class="st">"cm"</span>)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  }   </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="R_dada2_tutorial_files/figure-revealjs/fastq_quality-1.png" width="768"></p>
</div>
<div class="cell-output-display">
<p><img data-src="R_dada2_tutorial_files/figure-revealjs/fastq_quality-2.png" width="768"></p>
</div>
</div>
</section></section>
<section>
<section id="filter-and-trim-the-reads" class="title-slide slide level1 center">
<h1>Filter and Trim the reads</h1>

</section>
<section id="two-approaches" class="slide level2">
<h2>Two approaches</h2>
<p>The dada2 algorithm requires primers to be removed prior to processing.</p>
<ul>
<li>Using dada2 there are 2 possibilities
<ul>
<li>Remove by sequence, but dada2 does not allow for ambiguities</li>
<li>Remove by position, which is not a problem for Illumina sequences but is a problem for 454</li>
</ul></li>
<li>For complex situation we recommend to use <strong><a href="http://cutadapt.readthedocs.io/en/stable/guide.html">cutadapt</a></strong> to remove the primers.</li>
</ul>
<h3 id="create-names-for-the-filtered-files">Create names for the filtered files</h3>
<p>We create the name of the files that will be generated by the <code>filterAndTrim</code> function in the step below.</p>
<p>These names are composed by the path name (“../fastq_filtered/”), the sample names, the read number (R1 or R2) and a “_filt” suffix.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-11_abd5646ecfff092afb0fb148631738be">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>  filt_R1 <span class="ot">&lt;-</span> <span class="fu">str_c</span>(filtered_dir, sample.names, <span class="st">"_R1_filt.fastq"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  filt_R2 <span class="ot">&lt;-</span> <span class="fu">str_c</span>(filtered_dir, sample.names, <span class="st">"_R2_filt.fastq"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="method-1---removing-the-primers-by-sequence" class="slide level2">
<h2>Method 1 - Removing the primers by sequence</h2>
<p>(DO NOT EXECUTE THIS STEP)</p>
<p>The next piece of code could be used to remove the primers by <strong>sequence</strong>.</p>
<p>The dada2 package does not allow for primer degeneracy. Since our forward primer is degenerated at two positions, all four combinations need to be tested.</p>
<p>However it will be necessary to re-assemble after that the 4 fastQ files created (which has not to done).</p>
<p>So the better strategy is to remove primer by truncation (see next step).</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-12_e5963a9a54cd8baa90880ec5dcfb419e">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># On Windows set multithread=FALSE  </span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  out_all <span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="at">id=</span><span class="fu">length</span>(fns_R1))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>) {</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fns_R1, filt_R1, fns_R2, filt_R2, <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">250</span>,<span class="dv">240</span>), <span class="at">trimLeft =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="cn">Inf</span>, <span class="cn">Inf</span>), <span class="at">truncQ=</span><span class="dv">10</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>, <span class="at">primer.fwd =</span> primer_set_fwd[i], </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">compress=</span><span class="cn">FALSE</span>, <span class="at">multithread=</span><span class="cn">FALSE</span>) </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    out_all <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out_all, out)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(out_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="method-2---remove-primers-by-truncation-and-filter" class="slide level2">
<h2>Method 2 - Remove primers by truncation and filter</h2>
<p>Filter all sequences with N, truncate R2 to 240 bp</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-13_94bdfb867c006a93931de93717331427">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fns_R1, filt_R1, fns_R2, filt_R2, </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">250</span>,<span class="dv">240</span>), <span class="at">trimLeft =</span> <span class="fu">c</span>(primer_length_fwd,primer_length_rev),</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">truncQ=</span><span class="dv">10</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>,  </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">compress=</span><span class="cn">FALSE</span>, <span class="at">multithread=</span><span class="cn">FALSE</span>) </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> reads.in </th>
   <th style="text-align:right;"> reads.out </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 120p_S39_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 256 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 121p_S57_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 457 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 122p_S4_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 407 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 125p_S22_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 553 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 126p_S40_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 508 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 140p_S5_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 456 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 141p_S23_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 473 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 142p_S41_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 583 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 155p_S59_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 528 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 156p_S6_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 530 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 157p_S24_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 513 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 165p_S42_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 521 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 166p_S60_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 519 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 167p_S7_R1.subsample.fastq </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 572 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section></section>
<section>
<section id="dada2-processing" class="title-slide slide level1 center">
<h1>Dada2 processing</h1>

</section>
<section id="learn-error-rates" class="slide level2">
<h2>Learn error rates</h2>
<p>The error rates are plotted.</p>
<h4 id="r1">R1</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/error_rates_R1_cad785ad5769222c3a75de96e0f37bbf">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>  err_R1 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filt_R1, <span class="at">multithread=</span><span class="cn">FALSE</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotErrors</span>(err_R1, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="R_dada2_tutorial_files/figure-revealjs/error_rates_R1-1.png" width="768"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>1581480 total bases in 6876 reads from 14 samples will be used for learning the error rates.</code></pre>
</div>
</div>
<p><br></p>
<h4 id="r2">R2</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/error_rates_R2_bb0cc1204c7eecc6edfa4d800126901a">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>  err_R2 <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filt_R2, <span class="at">multithread=</span><span class="cn">FALSE</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotErrors</span>(err_R2, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="R_dada2_tutorial_files/figure-revealjs/error_rates_R2-1.png" width="768"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>1505844 total bases in 6876 reads from 14 samples will be used for learning the error rates.</code></pre>
</div>
</div>
</section>
<section id="dereplicate-the-reads" class="slide level2">
<h2>Dereplicate the reads</h2>
<h4 id="r1-1">R1</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-16_0e706f4fd341725682ccd7faf21b2fa3">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>  derep_R1 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filt_R1, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the derep-class objects by the sample names</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(derep_R1) <span class="ot">&lt;-</span> sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<h4 id="r2-1">R2</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-17_53137eef52d68e914145f5f2ad8d1ae5">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>  derep_R2 <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filt_R2, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the derep-class objects by the sample names</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(derep_R2) <span class="ot">&lt;-</span> sample.names </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sequence-variant-inference-algorithm-to-the-dereplicated-data" class="slide level2">
<h2>Sequence-variant inference algorithm to the dereplicated data</h2>
<h4 id="r1-2">R1</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-18_8df7ae042ea87b837bcc46b997c80050">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>    dada_R1 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derep_R1, <span class="at">err=</span>err_R1, <span class="at">multithread=</span><span class="cn">FALSE</span>, <span class="at">pool=</span><span class="cn">FALSE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    dada_R1[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Sample 1 - 256 reads in 93 unique sequences.
Sample 2 - 457 reads in 166 unique sequences.
Sample 3 - 407 reads in 128 unique sequences.
Sample 4 - 553 reads in 220 unique sequences.
Sample 5 - 508 reads in 219 unique sequences.
Sample 6 - 456 reads in 147 unique sequences.
Sample 7 - 473 reads in 180 unique sequences.
Sample 8 - 583 reads in 211 unique sequences.
Sample 9 - 528 reads in 172 unique sequences.
Sample 10 - 530 reads in 211 unique sequences.
Sample 11 - 513 reads in 177 unique sequences.
Sample 12 - 521 reads in 199 unique sequences.
Sample 13 - 519 reads in 172 unique sequences.
Sample 14 - 572 reads in 170 unique sequences.
dada-class: object describing DADA2 denoising results
5 sequence variants were inferred from 93 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
</div>
</div>
<p><br></p>
<h4 id="r2-2">R2</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-19_8fd146f278475d537f25880299357e44">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>    dada_R2 <span class="ot">&lt;-</span> <span class="fu">dada</span>(derep_R2, <span class="at">err=</span>err_R2, <span class="at">multithread=</span><span class="cn">FALSE</span>, <span class="at">pool=</span><span class="cn">FALSE</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>    dada_R2[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Sample 1 - 256 reads in 236 unique sequences.
Sample 2 - 457 reads in 391 unique sequences.
Sample 3 - 407 reads in 299 unique sequences.
Sample 4 - 553 reads in 409 unique sequences.
Sample 5 - 508 reads in 451 unique sequences.
Sample 6 - 456 reads in 335 unique sequences.
Sample 7 - 473 reads in 347 unique sequences.
Sample 8 - 583 reads in 458 unique sequences.
Sample 9 - 528 reads in 418 unique sequences.
Sample 10 - 530 reads in 404 unique sequences.
Sample 11 - 513 reads in 384 unique sequences.
Sample 12 - 521 reads in 395 unique sequences.
Sample 13 - 519 reads in 396 unique sequences.
Sample 14 - 572 reads in 400 unique sequences.
dada-class: object describing DADA2 denoising results
3 sequence variants were inferred from 236 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
</div>
</div>
</section>
<section id="merge-sequences" class="slide level2">
<h2>Merge sequences</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-20_54ba79968a69e726f91730bf9a43c186">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>    mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dada_R1, derep_R1, dada_R2, derep_R2, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Inspect the merger data.frame from the first sample</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  df_table <span class="ot">&lt;-</span> mergers[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sequence =</span> <span class="fu">str_c</span>(<span class="fu">str_sub</span>(sequence, <span class="dv">1</span>, <span class="dv">20</span>), <span class="st">"..."</span>))</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(df_table, <span class="st">"html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> sequence </th>
   <th style="text-align:right;"> abundance </th>
   <th style="text-align:right;"> forward </th>
   <th style="text-align:right;"> reverse </th>
   <th style="text-align:right;"> nmatch </th>
   <th style="text-align:right;"> nmismatch </th>
   <th style="text-align:right;"> nindel </th>
   <th style="text-align:right;"> prefer </th>
   <th style="text-align:left;"> accept </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> AGCTCCAATAGCGTATATTA... </td>
   <td style="text-align:right;"> 146 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 71 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:left;"> TRUE </td>
  </tr>
  <tr>
   <td style="text-align:left;"> CACACGTCTAATGTTGCATT... </td>
   <td style="text-align:right;"> 64 </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 131 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:left;"> TRUE </td>
  </tr>
  <tr>
   <td style="text-align:left;"> AGCTCCAATAGCGTATACTA... </td>
   <td style="text-align:right;"> 13 </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 72 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:left;"> TRUE </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="make-sequence-table" class="slide level2">
<h2>Make sequence table</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-21_b10d1f9ab43024f640ff756d7940aa42">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>    seqtab <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dim</span>(seqtab)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make a transposed of the seqtab to make it be similar to mothur database</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    t_seqtab <span class="ot">&lt;-</span> <span class="fu">t</span>(seqtab)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Inspect distribution of sequence lengths</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 14 58

318 351 360 363 367 369 371 372 373 375 376 377 378 379 380 382 383 384 387 388 
  3   1   1   3   1   1   2   1   2   1   3   6  12   4   4   2   5   2   1   3 </code></pre>
</div>
</div>
</section>
<section id="remove-chimeras" class="slide level2">
<h2>Remove chimeras</h2>
<p>Note that remove chimeras will produce spurious results if primers have not be removed. The parameter <code>methods</code> can be <code>pooled</code> or <code>consensus</code></p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-22_9e104460ebdd25fd8ac595dcf2b34889">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>    seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab, <span class="at">method=</span><span class="st">"consensus"</span>, <span class="at">multithread=</span><span class="cn">FALSE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Compute % of non chimeras</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"% of non chimeras : "</span>,<span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"total number of sequences : "</span>,<span class="fu">sum</span>(seqtab.nochim))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "% of non chimeras : 100"
[1] "total number of sequences : 5868"</code></pre>
</div>
</div>
<p>In our case there were no chimeras found. It is noteworthy that the total number of sequences is almost twice that what is recovered with <strong>mothur</strong> which is <strong>2573</strong></p>
</section>
<section id="track-number-of-reads-at-each-step" class="slide level2">
<h2>Track number of reads at each step</h2>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-23_1e111edda975ccfa2acb36b7cdf493cd">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># define a function</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out, <span class="fu">sapply</span>(dada_R1, getN), <span class="fu">sapply</span>(mergers, getN), </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">rowSums</span>(seqtab), <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"input"</span>, <span class="st">"filtered"</span>, <span class="st">"denoised"</span>, <span class="st">"merged"</span>, <span class="st">"tabled"</span>, <span class="st">"nonchim"</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    knitr<span class="sc">::</span><span class="fu">kable</span>(track)  </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(<span class="fu">data.frame</span>(track), <span class="fu">str_c</span>(dada2_dir,<span class="st">"read_numbers_dada2.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> input </th>
   <th style="text-align:right;"> filtered </th>
   <th style="text-align:right;"> denoised </th>
   <th style="text-align:right;"> merged </th>
   <th style="text-align:right;"> tabled </th>
   <th style="text-align:right;"> nonchim </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 120p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 256 </td>
   <td style="text-align:right;"> 241 </td>
   <td style="text-align:right;"> 223 </td>
   <td style="text-align:right;"> 223 </td>
   <td style="text-align:right;"> 223 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 121p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 457 </td>
   <td style="text-align:right;"> 446 </td>
   <td style="text-align:right;"> 397 </td>
   <td style="text-align:right;"> 397 </td>
   <td style="text-align:right;"> 397 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 122p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 407 </td>
   <td style="text-align:right;"> 396 </td>
   <td style="text-align:right;"> 357 </td>
   <td style="text-align:right;"> 357 </td>
   <td style="text-align:right;"> 357 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 125p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 553 </td>
   <td style="text-align:right;"> 551 </td>
   <td style="text-align:right;"> 464 </td>
   <td style="text-align:right;"> 464 </td>
   <td style="text-align:right;"> 464 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 126p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 508 </td>
   <td style="text-align:right;"> 493 </td>
   <td style="text-align:right;"> 340 </td>
   <td style="text-align:right;"> 340 </td>
   <td style="text-align:right;"> 340 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 140p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 456 </td>
   <td style="text-align:right;"> 441 </td>
   <td style="text-align:right;"> 427 </td>
   <td style="text-align:right;"> 427 </td>
   <td style="text-align:right;"> 427 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 141p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 473 </td>
   <td style="text-align:right;"> 460 </td>
   <td style="text-align:right;"> 381 </td>
   <td style="text-align:right;"> 381 </td>
   <td style="text-align:right;"> 381 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 142p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 583 </td>
   <td style="text-align:right;"> 567 </td>
   <td style="text-align:right;"> 495 </td>
   <td style="text-align:right;"> 495 </td>
   <td style="text-align:right;"> 495 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 155p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 528 </td>
   <td style="text-align:right;"> 524 </td>
   <td style="text-align:right;"> 445 </td>
   <td style="text-align:right;"> 445 </td>
   <td style="text-align:right;"> 445 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 156p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 530 </td>
   <td style="text-align:right;"> 525 </td>
   <td style="text-align:right;"> 425 </td>
   <td style="text-align:right;"> 425 </td>
   <td style="text-align:right;"> 425 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 157p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 513 </td>
   <td style="text-align:right;"> 507 </td>
   <td style="text-align:right;"> 438 </td>
   <td style="text-align:right;"> 438 </td>
   <td style="text-align:right;"> 438 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 165p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 521 </td>
   <td style="text-align:right;"> 509 </td>
   <td style="text-align:right;"> 442 </td>
   <td style="text-align:right;"> 442 </td>
   <td style="text-align:right;"> 442 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 166p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 519 </td>
   <td style="text-align:right;"> 510 </td>
   <td style="text-align:right;"> 478 </td>
   <td style="text-align:right;"> 478 </td>
   <td style="text-align:right;"> 478 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 167p </td>
   <td style="text-align:right;"> 1000 </td>
   <td style="text-align:right;"> 572 </td>
   <td style="text-align:right;"> 563 </td>
   <td style="text-align:right;"> 556 </td>
   <td style="text-align:right;"> 556 </td>
   <td style="text-align:right;"> 556 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="transforming-and-saving-the-asvs-sequences" class="slide level2">
<h2>Transforming and saving the ASVs sequences</h2>
<p>In the output of dada2, otu names are the sequences.</p>
<p>We change to give a Otuxxx name and the sequences are stored in the taxonomy table.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-24_8a6fa68659840bd43d252a00eb806f78">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>  seqtab.nochim_trans <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(seqtab.nochim)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sequence"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowid_to_column</span>(<span class="at">var =</span> <span class="st">"OTUNumber"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">remove_rownames</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">OTUNumber =</span> <span class="fu">sprintf</span>(<span class="st">"otu%04d"</span>, OTUNumber)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sequence =</span> <span class="fu">str_replace_all</span>(sequence, <span class="st">"(-|</span><span class="sc">\\</span><span class="st">.)"</span>,<span class="st">""</span>))</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> seqtab.nochim_trans</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  seq_out <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">DNAStringSet</span>(df<span class="sc">$</span>sequence)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(seq_out) <span class="ot">&lt;-</span> df<span class="sc">$</span>OTUNumber</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>  Biostrings<span class="sc">::</span><span class="fu">writeXStringSet</span>(seq_out, <span class="fu">str_c</span>(dada2_dir, <span class="st">"CARBOM_ASV_no_taxo.fasta"</span>), </span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">compress=</span><span class="cn">FALSE</span>, <span class="at">width =</span> <span class="dv">20000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="assigning-taxonomy" class="slide level2">
<h2>Assigning taxonomy</h2>
<p>This step is quite long… If you want to skip please go to next step.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/assign_taxa_353ea0029324c7612124228b8c7c9e30">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>    pr2_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(database_dir, <span class="st">"pr2_version_4.14.0_SSU_dada2.fasta.gz"</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    taxa <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="at">refFasta=</span>pr2_file,  </span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">taxLevels =</span> PR2_tax_levels,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">minBoot =</span> <span class="dv">0</span>, <span class="at">outputBootstraps =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(taxa, <span class="fu">str_c</span>(dada2_dir, <span class="st">"CARBOM.taxa.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section>
<section>
<section id="export-data" class="title-slide slide level1 center">
<h1>Export data</h1>

</section>
<section id="export" class="slide level2">
<h2>Export</h2>
<p>We need to reformat the data produced by dada2 if we want to use for further analysis, for example with Phyloseq or BLAST.</p>
<h4 id="export-data-as-produced-by-dada2">Export data as produced by Dada2</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-26_3b788243dfb976c8499a60a6763817f1">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>  taxa <span class="ot">&lt;-</span>  <span class="fu">readRDS</span>(<span class="fu">str_c</span>(dada2_dir, <span class="st">"CARBOM.taxa.rds"</span>))  </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="fu">as.tibble</span>(taxa<span class="sc">$</span>tax), <span class="at">path =</span> <span class="fu">str_c</span>(dada2_dir, <span class="st">"taxa.txt"</span>))  </span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="fu">as.tibble</span>(taxa<span class="sc">$</span>boot), <span class="at">path =</span> <span class="fu">str_c</span>(dada2_dir, <span class="st">"taxa_boot.txt"</span>))</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="fu">as.tibble</span>(seqtab.nochim), <span class="at">path =</span> <span class="fu">str_c</span>(dada2_dir, <span class="st">"seqtab.txt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<h4 id="appending-taxonomy-and-boot-to-the-sequence-table">Appending taxonomy and boot to the sequence table</h4>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-27_215a5bc536dc2b3678fb7f136582fdeb">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>    taxa_tax <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(taxa<span class="sc">$</span>tax)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>    taxa_boot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(taxa<span class="sc">$</span>boot) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename_all</span>(<span class="fu">funs</span>(<span class="fu">str_c</span>(.,<span class="st">"_boot"</span>)))</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    seqtab.nochim_trans <span class="ot">&lt;-</span> seqtab.nochim_trans <span class="sc">%&gt;%</span> </span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_cols</span>(taxa_tax) <span class="sc">%&gt;%</span> </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_cols</span>(taxa_boot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filter-for-18s" class="slide level2">
<h2>Filter for 18S</h2>
<p>Remember that we sequenced 3 genes (18S, 16S plastid and <em>nifH</em>).</p>
<p>We remove the sequences that are not 18S by selecting only bootstrap values for Supergroup in excess of 80.</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-28_8c95fffa8a28b9d66112f134c4733a87">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>    bootstrap_min <span class="ot">&lt;-</span> <span class="dv">80</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter based on the bootstrap</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    seqtab.nochim_18S <span class="ot">&lt;-</span> seqtab.nochim_trans <span class="sc">%&gt;%</span>  </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(Supergroup_boot <span class="sc">&gt;=</span> bootstrap_min) </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a database like file for dada2</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(seqtab.nochim_18S, <span class="fu">str_c</span>(dada2_dir, <span class="st">"CARBOM_dada2.database.tsv"</span>))</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Before filtration - # of sequences: "</span>, <span class="fu">nrow</span>(seqtab.nochim_trans), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"After filtration - # of sequences: "</span>, <span class="fu">nrow</span>(seqtab.nochim_18S), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Before filtration - # of sequences:  58 
After filtration - # of sequences:  53 </code></pre>
</div>
</div>
</section>
<section id="write-fasta-file-for-blast-analysis-with-taxonomy" class="slide level2">
<h2>Write FASTA file for BLAST analysis with taxonomy</h2>
<p>Use the Biostrings library</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-29_00dbe46f91c48cff23067e813c6d39c1">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> seqtab.nochim_18S</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  seq_out <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">DNAStringSet</span>(df<span class="sc">$</span>sequence)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(seq_out) <span class="ot">&lt;-</span> <span class="fu">str_c</span>(df<span class="sc">$</span>OTUNumber,</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Supergroup,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Division,</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Class,</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Order,</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Family,</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Genus,</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>                                df<span class="sc">$</span>Species,</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>                                <span class="at">sep=</span><span class="st">"|"</span>)</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  Biostrings<span class="sc">::</span><span class="fu">writeXStringSet</span>(seq_out, <span class="fu">str_c</span>(blast_dir, <span class="st">"CARBOM_ASV.fasta"</span>), </span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>                              <span class="at">compress=</span><span class="cn">FALSE</span>, <span class="at">width =</span> <span class="dv">20000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="write-fasta-file-for-blast-analysis-with-taxonomy-1" class="slide level2">
<h2>Write FASTA file for BLAST analysis with taxonomy</h2>
<p>This file can be sent to a server and a BLAST analysis can be done using the following bash file</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-30_c6298fb97c39765d44996a7f3df89a58">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace the next line by the location of the directory where you have your data</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_PROJECT</span><span class="op">=</span><span class="st">"/projet/sbr/ccebarcodep1408/workshop_nz_2018/blast/"</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$DIR_PROJECT</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="va">FILE</span><span class="op">=</span><span class="st">"CARBOM_ASV"</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="va">FASTA</span><span class="op">=</span><span class="va">$DIR_PROJECT$FILE</span><span class="st">".fasta"</span></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a><span class="va">BLAST_TSV</span><span class="op">=</span><span class="va">$DIR_PROJECT$FILE</span><span class="st">".blast.tsv"</span></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="va">OUT_FMT</span><span class="op">=</span><span class="st">"6 qseqid sseqid sacc stitle sscinames staxids sskingdoms sblastnames pident slen length mismatch gapopen qstart qend sstart send evalue bitscore"</span></span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a><span class="ex">blastn</span> <span class="at">-max_target_seqs</span> 100 <span class="at">-evalue</span> 1.00e-10 <span class="at">-query</span> <span class="va">$FASTA</span> <span class="at">-out</span> <span class="va">$BLAST_TSV</span> <span class="at">-db</span> /db/blast/all/nt <span class="at">-outfmt</span> <span class="st">"</span><span class="va">$OUT_FMT</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="phyloseq" class="slide level2">
<h2>Phyloseq</h2>
<p>Create and save a phyloseq object from dada2 results</p>
<div class="cell" data-hash="R_dada2_tutorial_cache/revealjs/unnamed-chunk-31_ca448a10853ccb9547a89fce114b2f93">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_name=</span>sample.names)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(samdf) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>OTU <span class="ot">&lt;-</span> seqtab.nochim_18S <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">remove_rownames</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"OTUNumber"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select_if</span>(is.numeric) <span class="sc">%&gt;%</span> </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">"_boot"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">otu_table</span>(<span class="at">taxa_are_rows=</span><span class="cn">TRUE</span>) </span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>TAX <span class="ot">&lt;-</span> seqtab.nochim_18S <span class="sc">%&gt;%</span> </span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">remove_rownames</span>()<span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"OTUNumber"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Kingdom<span class="sc">:</span>Species)<span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span>  </span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tax_table</span>()</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>ps_dada2 <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU, </span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sample_data</span>(samdf), </span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>               TAX)</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_dada2,<span class="fu">str_c</span>(dada2_dir, <span class="st">"CARBOM_phyloseq.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="footer footer-default">

</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="R_dada2_tutorial_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="R_dada2_tutorial_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: true,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1400,

        height: 1000,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>